core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -1742,4 +1753,129 @@ public void testCatalogTokenRefreshDisabledWithCredential() {
             eq(catalogHeaders),
             any());
   }
+
+  @Test
+  public void multipleDiffsAgainstSingleTable() {
```

Comment:
This isn't testing multiple changes. It's testing just one set. The fact that the `Transaction` accumulates changes across table operations isn't really material to the test.

I'd change this to `testSingleTable`.